src/Connection/MessageReceivers/BaseMessageReceiver.ts

```diff
@@ -0,0 +1,141 @@
+import {
+    ActionCode,
+    CallbackList,
+    CommunicationWrapper,
+    TouchFreeRequest,
+    TouchFreeRequestCallback,
+    WebSocketResponse,
+} from '../TouchFreeServiceTypes';
+import { IBaseMessageReceiver } from './IBaseMessageReceiver';
+
+/**
+ * Base message receiver class to contain generic message processing functionality
+ *
+ * @internal
+ */
+export abstract class BaseMessageReceiver<TMessage> implements IBaseMessageReceiver {
+    /**
+     * Constructs the class
+     *
+     * @param useQueue - Whether the message receiver should use a queue or just store the last item
+     */
+    constructor(useQueue: boolean) {
+        this.useQueue = useQueue;
```

Comment:
👀 Could change constructor to 

```
    constructor(useQueue: boolean, callbackHandler?: CallbackHandler) {
        this.useQueue = useQueue;
        this.setup(() => this.checkForState(callbackHandler));
    }
```

and add an abstract checkForState for the message receivers to overwrite. Then can remove the constructors from message receivers

Reply:
I'm 🤷 so if you didn't want to do it then happy with that decision